### DISCUSSION TITLE
throw redirect throws error

### DISCUSSION THREAD

**Discussion by LarsGIF** [sveltejs/kit#10098] [Q&A] [answered]
I my login module I have this code snippet in my +page.server.ts
```
 try {
      // Validate with Lucia
      const key = await auth.useKey("email", email, password);

      const session = await auth.createSession(key.userId);
      locals.auth.setSession(session);

      // Goto home page
      throw redirect(302, "/");
      
    } catch (error) {
```
When I single step in the debugger and step over "throw redirect", it will go into the catch block instead of redirecting.
I have seen this issue reported before for earlier versions of SvelteKit but it should have been resolved.
I am using 
``` 
"@sveltejs/kit": "^1.20.1",
"svelte": "^3.59.1",
```
so unless I have done something wrong somewhere, I think the problem still exists.

There are more places where throw redirect have the same behavior.

Is there anyone that have the same experience or any ideas on what could cause this problem?

**Comment by Renkas** [comment, marked as answer]
It seems to work as intended. You throw something in try block and then catch block catches it.

Why would you even want to throw redirect in a try block? What's the use case?

**Comment by finkel** [comment]
I am seeing this bug resurface as well. The temporary fix that I read about seems usually work: stop your server, delete the node-modules directory, and then start it up.
